build_runner/lib/src/generate/build_impl.dart

```diff
@@ -50,15 +51,17 @@ Future<BuildResult> build(List<BuildAction> buildActions,
     bool skipBuildScriptCheck,
     bool enableLowResourcesMode}) async {
   var options = new BuildOptions(
-      deleteFilesByDefault: deleteFilesByDefault,
-      writeToCache: writeToCache,
+      deleteFilesByDefault: deleteFilesByDefault ?? writeToCache,
       packageGraph: packageGraph,
       reader: reader,
       writer: writer,
       logLevel: logLevel,
       onLog: onLog,
       skipBuildScriptCheck: skipBuildScriptCheck,
       enableLowResourcesMode: enableLowResourcesMode);
+  if (writeToCache ?? false) {
```

Comment:
`== true` is more intuitive imo

Reply:
`== true` can cover either null or non-boolean types. I like `?? false` when I'm explicitly only worried about null but I know its a boolean.

I don't have strong feelings though, and this is going to go away soon so I changed it :)